pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -283,6 +290,19 @@ func getCABundle(ctx context.Context) *string {
 	return ptr.String(base64.StdEncoding.EncodeToString(transportConfig.TLS.CAData))
 }
 
+func kubeDNSIPv6(ctx context.Context, clientSet *kubernetes.Clientset) (net.IP, error) {
+	dnsService, err := clientSet.CoreV1().Services("kube-system").Get(ctx, "kube-dns", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	kubeDNSIP := net.ParseIP(dnsService.Spec.ClusterIP)
+	logging.FromContext(ctx).Debugf("Discovered DNS IP %s", kubeDNSIP)
+	if kubeDNSIP.To4() == nil { // ipv6
+		return kubeDNSIP, nil
+	}
+	return nil, nil
```

Comment:
Seems a bit odd to me to return nil, and then check nil later. Maybe better to just set the kubednsIP and then check if it's IPv6 vs IPv4 in the places we care.